### ISSUE TITLE
Unexpected parse error in for loop

### ISSUE THREAD

**Issue by dcrosa** [koalaman/shellcheck#8]
This parse error doesn't seem right:

In test.sh line 3:
for (( i=0 ; i<10 ; i++ )) ; do
^-- Couldn't parse this for loop.
                           ^-- Expected 'do'.
                           ^-- Unexpected ";". Fix any mentioned problems and try again.

Apparently, the parser is not expecting the semicolon before 'do', but man bash says it is a correct syntax:

```
   for (( expr1 ; expr2 ; expr3 )) ; do list ; done
          First, the arithmetic expression expr1 is evaluated according to the rules described below under ARITHMETIC EVALUATION.  The arithmetic expression expr2 is then evaluated repeatedly until it  evaluates  to
          zero.  Each time expr2 evaluates to a non-zero value, list is executed and the arithmetic expression expr3 is evaluated.  If any expression is omitted, it behaves as if it evaluates to 1.  The return value
          is the exit status of the last command in list that is executed, or false if any of the expressions is invalid.
```
